parser/lexer.go

```diff
@@ -104,6 +108,11 @@ func (s *Scanner) Lex(v *yySymType) int {
 			tok = tok1
 		}
 	}
+	if (s.sqlMode&mysql.ModeANSIQuotes) > 0 &&
```

Comment:
I think identifier within "" would always be a kind of identifier token.
This is the difference: select will be rewrite to kind of token named select, while `select` is always remain a identifier token.
In ANSI_QUOTES mode, "select" behavior is similar to `select`, so no rewrite is needed (line 104).